src/transaction_builder.js

```diff
@@ -524,10 +524,19 @@ TransactionBuilder.prototype.setVersion = function (version, overwinter = true)
       throw new Error('Unsupported Zcash transaction')
     }
     this.tx.overwintered = (overwinter ? 1 : 0)
+    this.tx.consensusBranchId = this.network.consensusBranchId[version]
   }
   this.tx.version = version
 }
 
+TransactionBuilder.prototype.setConsensusBranchId = function (consensusBranchId) {
+  if (!coins.isZcash(this.network)) {
+    throw new Error('consensusBranchId can only be set for Zcash transactions')
+  }
```

Comment:
```suggestion
  }
  if (!this.inputs.every(function (input) { return input.signatures === undefined })) {
    throw new Error('No, this would invalidate signatures')
  }
```

Reply:
Excellent thank you! updated in force-push to ed840b829079827c8d19458b4464e3a8ee498e90

Reply:
I'm actually thinking of pushing back on this and removing the check. There's a few places downstream where this library is used to sign multisig in two stages. Both stages call `setConsensusBranchId()` to override the default, and the second one is failing because there's already one signature in place. If a caller does screw this up, they'll find out when they try to broadcast that one sig had the wrong `consensusBranchId`. I suppose a super advanced method would actually check the incoming function argument against any existing signatures to make sure we aren't mixing values up. This is a little different than changing the `locktime` for example if a tx already has a sig. In this case we aren't really invalidating the original sig, we're just possibly creating a transaction that won't pass consensus checks.